### ISSUE TITLE
Unable to seed the DB

### ISSUE THREAD

**Issue by aaryandewan** [vercel/next-learn#868]
Been following this tutorial: https://nextjs.org/learn/dashboard-app/setting-up-your-database

When I'm trying to see the DB, it says:
![image](https://github.com/user-attachments/assets/aab7edb1-51e2-47c3-81d5-be8439ca89a4)

I don't know what to do. I have tried everything. 

My route.ts in my seed folder looks like: 

![image](https://github.com/user-attachments/assets/7400423e-9d88-459c-a756-d70bedb93e32)

My .env is also pretty good. it's set.

**Comment by arytix** [comment]
Same issue here, been trying this since yesterday. I commented everything out in the route.ts file except the import statements and the `const client = await db.connect();`
This statement raises the error shown above.  So, there seems to be some sort of issue with the db object imported from @vercel/postgres.
I also tried multiple version of node, react, next etc. but always resulting in the same error. Interestengly it works when deployed and the seed page is opened on the deployed instance. But that doesn't help much as the forthcoming chapters rely on a local connection to the database. So fixing this issue is essential for the rest of the tutorial.

**Comment by aaryandewan** [comment]
@arytix
 
Do this
1) Make sure that you have uncommented out the seed/route.ts file
2) You need to deploy your project on vercel.com
3) Go to this url in your browser (whatever your deployed app's URL is obviously not this exact URL)
![image](https://github.com/user-attachments/assets/8f0d6c74-d668-4888-9d9f-6f8b64e84537)
4) Go to it's /seed. Do not go to localhost:3000/seed, instead go to "https://next-learn-o9ve.vercel.app/seed" **[GO TO YOUR PROJECT's URL]**

**Comment by arytix** [comment]
Yes, maybe if you read my comment completely you would see that futher down the road you still keep on having issues connecting to the database.

**Comment by arytix** [comment]
If you continue to chapter 7, where you start fetching data. It also won't work locally.
![image](https://github.com/user-attachments/assets/6ecd8770-7e1a-4c27-b00a-c6dc0325d91c)

**Comment by arytix** [comment]
It got it working. I deleted everything and rebuilt from scratch, but this time I choose the server location of the DB closer to my physical presence instead of leaving it at the default of Washington. And now everything wondrously works. There seems to be some sort of latency/distance restriction.

**Comment by sandeeppal007** [comment]
paste this in route.ts
update this push this code on  githhub then go the 



import bcrypt from 'bcrypt';
import { db } from '@vercel/postgres';
import { invoices, customers, revenue, users } from '../lib/placeholder-data';

const client = await db.connect();

async function seedUsers() {
  await client.sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;
  await client.sql`
    CREATE TABLE IF NOT EXISTS users (
      id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
      name VARCHAR(255) NOT NULL,
      email TEXT NOT NULL UNIQUE,
      password TEXT NOT NULL
    );
  `;

  const insertedUsers = await Promise.all(
    users.map(async (user) => {
      const hashedPassword = await bcrypt.hash(user.password, 10);
      return client.sql`
        INSERT INTO users (id, name, email, password)
        VALUES (${user.id}, ${user.name}, ${user.email}, ${hashedPassword})
        ON CONFLICT (id) DO NOTHING;
      `;
    }),
  );

  return insertedUsers;
}

async function seedInvoices() {
  await client.sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;

  await client.sql`
    CREATE TABLE IF NOT EXISTS invoices (
      id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
      customer_id UUID NOT NULL,
      amount INT NOT NULL,
      status VARCHAR(255) NOT NULL,
      date DATE NOT NULL
    );
  `;

  const insertedInvoices = await Promise.all(
    invoices.map(
      (invoice) => client.sql`
        INSERT INTO invoices (customer_id, amount, status, date)
        VALUES (${invoice.customer_id}, ${invoice.amount}, ${invoice.status}, ${invoice.date})
        ON CONFLICT (id) DO NOTHING;
      `,
    ),
  );

  return insertedInvoices;
}

async function seedCustomers() {
  await client.sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;

  await client.sql`
    CREATE TABLE IF NOT EXISTS customers (
      id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
      name VARCHAR(255) NOT NULL,
      email VARCHAR(255) NOT NULL,
      image_url VARCHAR(255) NOT NULL
    );
  `;

  const insertedCustomers = await Promise.all(
    customers.map(
      (customer) => client.sql`
        INSERT INTO customers (id, name, email, image_url)
        VALUES (${customer.id}, ${customer.name}, ${customer.email}, ${customer.image_url})
        ON CONFLICT (id) DO NOTHING;
      `,
    ),
  );

  return insertedCustomers;
}

async function seedRevenue() {
  await client.sql`
    CREATE TABLE IF NOT EXISTS revenue (
      month VARCHAR(4) NOT NULL UNIQUE,
      revenue INT NOT NULL
    );
  `;

  const insertedRevenue = await Promise.all(
    revenue.map(
      (rev) => client.sql`
        INSERT INTO revenue (month, revenue)
        VALUES (${rev.month}, ${rev.revenue})
        ON CONFLICT (month) DO NOTHING;
      `,
    ),
  );

  return insertedRevenue;
}

export async function GET() {
 
  try {
    await client.sql`BEGIN`;
    await seedUsers();
    await seedCustomers();
    await seedInvoices();
    await seedRevenue();
    await client.sql`COMMIT`;

    return Response.json({ message: 'Database seeded successfully' });
  } catch (error) {
    await client.sql`ROLLBACK`;
    return Response.json({ error }, { status: 500 });
  }
}

**Comment by anasahammad** [comment]
When i try to install vercel/postgresql it occurs this error
![error](https://github.com/user-attachments/assets/33aa7ff7-9514-46ad-9b37-c3f8d4799e6d)

I cannot see the message of Database seeded successfully,

**Comment by siddhesh-k** [comment]
After uncommenting the route.ts file remove this 
```
return Response.json({
    message:
      'Uncomment this file and remove this line. You can delete this file when you are finished.',
  });
```
from line 105 to 108
and run
http://localhost:3000/seed
this is enough to get you seeds load u[p in your db

**Comment by Angeluz88** [comment]
Hello guys, the solution is as simple as deleting from line 105 to 108

**Comment by SteaneMurphy** [comment]
Honestly the amount of people in these forums who are so confident in their incorrect answers is astounding. The issue, if any of you morons could learn to read properly, is that running the /seed endpoint locally throws an 'object [Object]' error. This has nothing to do with deleting lines 105 - 108. It is also not the correct solution to run the fucking development project in production, for fucks sake. Would you actually test and run development code direct to prod?

The issue seems to be related to the @vercel/postgres package. So far the only thing that has fixed this issue, is to not use Vercel's shit postgres package and use Node's version instead.

**Comment by SteaneMurphy** [comment]
The issue as far as I can tell is that the Washington default server causes a latency that lags out the request. If you are using the "@vercel/postgres" package to connect to their cloud database, it might work better if you delete the database, re-create the database, and select a server close to your geographic location.

One of the ways I confirmed this was the issue:

- used the cloud deployed site with the /seed endpoint, this successfully seeded the database (Washington server)
- created a new database with a geographically close server to me, this also worked from the deployed /seed endpoint (Closer server)
- the new database also worked locally when using the local /seed endpoint (Closer server)

Without looking into the specific code that is a part of the Vercel postgres package, I am going to assume that the request is timing out (read: unresolved promise) due to geographic location. This issue should be fixed if someone from Vercel ever has 20 minutes to fix this.

**Comment by leerob** [comment]
https://github.com/vercel/next-learn/pull/989
